{
  "type": "issue",
  "issue": {
    "id": 2246583751,
    "node_id": "I_kwDOABII586F6CXH",
    "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/29897",
    "repository_url": "https://api.github.com/repos/bitcoin/bitcoin",
    "labels_url": "https://api.github.com/repos/bitcoin/bitcoin/issues/29897/labels%7B/name%7D",
    "comments_url": "https://api.github.com/repos/bitcoin/bitcoin/issues/29897/comments",
    "events_url": "https://api.github.com/repos/bitcoin/bitcoin/issues/29897/events",
    "html_url": "https://github.com/bitcoin/bitcoin/issues/29897",
    "number": 29897,
    "state": "open",
    "state_reason": null,
    "title": "Intermittent issue in p2p_unrequested_blocks.py\", line 98, in run_test                                        test_node.send_and_ping(msg_block(blocks_h2[0]))  assert self.is_connected                                    AssertionError / AcceptBlock FAILED (time-too-old, block's timestamp is too early) ",
    "body": "https://github.com/bitcoin/bitcoin/actions/runs/8689751788/job/23828088498?pr=29877#step:27:584\r\n\r\n```\r\nnode0 2024-04-15T13:27:36.008782Z [D:\\a\\bitcoin\\bitcoin\\src\\net_processing.cpp:4743] [ProcessMessage] [net] received block 20076ff9e9d9bace59d7621747b34dfb4af9b2f75ff499017397854f9fefb930 peer=0 \r\n node0 2024-04-15T13:27:36.008822Z [D:\\a\\bitcoin\\bitcoin\\src\\validation.cpp:4145] [AcceptBlockHeader] [validation] AcceptBlockHeader: Consensus::ContextualCheckBlockHeader: 20076ff9e9d9bace59d7621747b34dfb4af9b2f75ff499017397854f9fefb930, time-too-old, block's timestamp is too early \r\n node0 2024-04-15T13:27:36.008857Z [D:\\a\\bitcoin\\bitcoin\\src\\validationinterface.cpp:245] [BlockChecked] [validation] BlockChecked: block hash=20076ff9e9d9bace59d7621747b34dfb4af9b2f75ff499017397854f9fefb930 state=time-too-old, block's timestamp is too early \r\n node0 2024-04-15T13:27:36.008883Z [D:\\a\\bitcoin\\bitcoin\\src\\net_processing.cpp:1827] [Misbehaving] [net] Misbehaving: peer=0 (0 -> 100) DISCOURAGE THRESHOLD EXCEEDED \r\nError: 2024-04-15T13:27:36.008905Z [D:\\a\\bitcoin\\bitcoin\\src\\validation.cpp:4396] [ProcessNewBlock] [error] ProcessNewBlock: AcceptBlock FAILED (time-too-old, block's timestamp is too early) \r\n node0 2024-04-15T13:27:36.008927Z [D:\\a\\bitcoin\\bitcoin\\src\\net_processing.cpp:5067] [MaybeDiscourageAndDisconnect] [net] Warning: disconnecting but not discouraging local peer 0! \r\n node0 2024-04-15T13:27:36.009001Z [D:\\a\\bitcoin\\bitcoin\\src\\net.cpp:541] [CloseSocketDisconnect] [net] disconnecting peer=0 \r\n node0 2024-04-15T13:27:36.009142Z [D:\\a\\bitcoin\\bitcoin\\src\\net_processing.cpp:1708] [FinalizeNode] [net] Cleared nodestate for peer=0 \r\n test  2024-04-15T13:27:36.058000Z TestFramework (ERROR): Assertion failed \r\n                                   Traceback (most recent call last):\r\n                                     File \"D:\\a\\bitcoin\\bitcoin\\test\\functional\\test_framework\\test_framework.py\", line 132, in main\r\n                                       self.run_test()\r\n                                     File \"D:\\a\\bitcoin\\bitcoin/test/functional/p2p_unrequested_blocks.py\", line 98, in run_test\r\n                                       test_node.send_and_ping(msg_block(blocks_h2[0]))\r\n                                     File \"D:\\a\\bitcoin\\bitcoin\\test\\functional\\test_framework\\p2p.py\", line 686, in send_and_ping\r\n                                       self.sync_with_ping(timeout=timeout)\r\n                                     File \"D:\\a\\bitcoin\\bitcoin\\test\\functional\\test_framework\\p2p.py\", line 699, in sync_with_ping\r\n                                       self.wait_until(test_function, timeout=timeout)\r\n                                     File \"D:\\a\\bitcoin\\bitcoin\\test\\functional\\test_framework\\p2p.py\", line 586, in wait_until\r\n                                       wait_until_helper_internal(test_function, timeout=timeout, lock=p2p_lock, timeout_factor=self.timeout_factor)\r\n                                     File \"D:\\a\\bitcoin\\bitcoin\\test\\functional\\test_framework\\util.py\", line 279, in wait_until_helper_internal\r\n                                       if predicate():\r\n                                          ^^^^^^^^^^^\r\n                                     File \"D:\\a\\bitcoin\\bitcoin\\test\\functional\\test_framework\\p2p.py\", line 583, in test_function\r\n                                       assert self.is_connected\r\n                                   AssertionError",
    "user": {
      "login": "maflcko",
      "id": 6399679,
      "node_id": "MDQ6VXNlcjYzOTk2Nzk=",
      "avatar_url": "https://avatars.githubusercontent.com/u/6399679?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/maflcko",
      "html_url": "https://github.com/maflcko",
      "followers_url": "https://api.github.com/users/maflcko/followers",
      "following_url": "https://api.github.com/users/maflcko/following%7B/other_user%7D",
      "gists_url": "https://api.github.com/users/maflcko/gists%7B/gist_id%7D",
      "starred_url": "https://api.github.com/users/maflcko/starred%7B/owner%7D%7B/repo%7D",
      "subscriptions_url": "https://api.github.com/users/maflcko/subscriptions",
      "organizations_url": "https://api.github.com/users/maflcko/orgs",
      "repos_url": "https://api.github.com/users/maflcko/repos",
      "events_url": "https://api.github.com/users/maflcko/events%7B/privacy%7D",
      "received_events_url": "https://api.github.com/users/maflcko/received_events",
      "type": "User",
      "site_admin": false
    },
    "labels": [
      {
        "id": 62963516,
        "node_id": "MDU6TGFiZWw2Mjk2MzUxNg==",
        "url": "https://api.github.com/repos/bitcoin/bitcoin/labels/Tests",
        "name": "Tests",
        "color": "d4c5f9",
        "default": false
      },
      {
        "id": 5334691551,
        "node_id": "LA_kwDOABII588AAAABPfju3w",
        "url": "https://api.github.com/repos/bitcoin/bitcoin/labels/CI%20failed",
        "name": "CI failed",
        "description": "",
        "color": "cccccc",
        "default": false
      }
    ],
    "assignees": [],
    "author_association": "MEMBER",
    "locked": false,
    "comments": 0,
    "created_at": "2024-04-16T17:44:57Z",
    "updated_at": "2024-04-16T17:45:08Z"
  },
  "events": [
    {
      "event": "labeled",
      "id": 12494056097,
      "node_id": "LE_lADOABII586F6CXHzwAAAALotCqh",
      "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/events/12494056097",
      "actor": {
        "login": "maflcko",
        "id": 6399679,
        "node_id": "MDQ6VXNlcjYzOTk2Nzk=",
        "avatar_url": "https://avatars.githubusercontent.com/u/6399679?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/maflcko",
        "html_url": "https://github.com/maflcko",
        "followers_url": "https://api.github.com/users/maflcko/followers",
        "following_url": "https://api.github.com/users/maflcko/following%7B/other_user%7D",
        "gists_url": "https://api.github.com/users/maflcko/gists%7B/gist_id%7D",
        "starred_url": "https://api.github.com/users/maflcko/starred%7B/owner%7D%7B/repo%7D",
        "subscriptions_url": "https://api.github.com/users/maflcko/subscriptions",
        "organizations_url": "https://api.github.com/users/maflcko/orgs",
        "repos_url": "https://api.github.com/users/maflcko/repos",
        "events_url": "https://api.github.com/users/maflcko/events%7B/privacy%7D",
        "received_events_url": "https://api.github.com/users/maflcko/received_events",
        "type": "User",
        "site_admin": false
      },
      "commit_id": null,
      "commit_url": null,
      "created_at": "2024-04-16T17:45:07Z",
      "label": {
        "name": "Tests",
        "color": "d4c5f9"
      }
    },
    {
      "event": "labeled",
      "id": 12494056105,
      "node_id": "LE_lADOABII586F6CXHzwAAAALotCqp",
      "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/events/12494056105",
      "actor": {
        "login": "maflcko",
        "id": 6399679,
        "node_id": "MDQ6VXNlcjYzOTk2Nzk=",
        "avatar_url": "https://avatars.githubusercontent.com/u/6399679?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/maflcko",
        "html_url": "https://github.com/maflcko",
        "followers_url": "https://api.github.com/users/maflcko/followers",
        "following_url": "https://api.github.com/users/maflcko/following%7B/other_user%7D",
        "gists_url": "https://api.github.com/users/maflcko/gists%7B/gist_id%7D",
        "starred_url": "https://api.github.com/users/maflcko/starred%7B/owner%7D%7B/repo%7D",
        "subscriptions_url": "https://api.github.com/users/maflcko/subscriptions",
        "organizations_url": "https://api.github.com/users/maflcko/orgs",
        "repos_url": "https://api.github.com/users/maflcko/repos",
        "events_url": "https://api.github.com/users/maflcko/events%7B/privacy%7D",
        "received_events_url": "https://api.github.com/users/maflcko/received_events",
        "type": "User",
        "site_admin": false
      },
      "commit_id": null,
      "commit_url": null,
      "created_at": "2024-04-16T17:45:07Z",
      "label": {
        "name": "CI failed",
        "color": "cccccc"
      }
    }
  ]
}